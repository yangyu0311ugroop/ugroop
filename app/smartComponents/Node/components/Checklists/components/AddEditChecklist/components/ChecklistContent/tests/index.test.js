import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import { ChecklistContent } from '../index';

describe('<ChecklistContent />', () => {
  let rendered;
  let instance;

  const resaga = {
    setValue: jest.fn(),
    dispatchTo: jest.fn(),
  };

  const props = {
    classes: {},
    resaga,
  };

  beforeEach(() => {
    rendered = shallow(<ChecklistContent {...props} />);
    instance = rendered.instance();
  });

  it('should exists', () => {
    expect(ChecklistContent).toBeDefined();
  });

  it('should render without exploding', () => {
    expect(rendered.length).toBe(1);
  });

  describe('render()', () => {
    it('should render correctly', () => {
      rendered.setProps({ index: 2, id: 1 });

      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
    it('should render correctly', () => {
      rendered.setProps({ rootNodeId: 22999, parentNodeId: 1 });

      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
});
