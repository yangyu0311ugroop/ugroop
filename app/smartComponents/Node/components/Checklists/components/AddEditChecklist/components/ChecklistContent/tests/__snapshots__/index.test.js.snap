// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChecklistContent /> render() should render correctly 1`] = `<div />`;

exports[`<ChecklistContent /> render() should render correctly 2`] = `
<div>
  <ConnectFunction
    parentNodeId={1}
    variant="textOnly"
  />
</div>
`;
